### ISSUE TITLE
Problem with game scheme settings: If AI team wins before it ends turn, the game crashes

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#139]
```
What steps will reproduce the problem?
1. Create game scheme with following game modifiers:
-Solid land
-Reset health
-Disable land objects
-Unlimited Attacks
-Per Hedgehog Ammo
And following basic settings (I don't thing they cause the problem):
-Damage 300, Turn time 15, Initial Health 100
-Sudden death: timeout 5, water rise 0, health decrease 5
-Rope Length 100%, Crate drops never, Health crates 25%, Health in crates 25,
-Mines time 0 sec, Mines 12, Dud mines 0% and 24 explosives.
2. Run fight with at least 1 AI team.
3. Let any of the AI teams win and make sure the turn time isn't over.

What is the expected output? What do you see instead?
The game exits to game settings menu without showing info about winning team 
and round summary. I haven't checked, if the same thing happens when all 
hedgehogs die before turn ends. The reason of such behavior isn't related to 
Unlimited Attacks modifier, because the problem doesn't occur in Clean Slate 
scheme, which uses the modifier.

What version of the product are you using? On what operating system?
0.9.15 on Windows XP SP2

Please provide any additional information below.
```

-----
Original issue reported on code.google.com by `adibiaz...@gmail.com` on 31 Dec 2010 at 5:01

**Comment by GoogleCodeExporter** [comment]
```
With "normal" game (except: turn time: 300s), too.
The AI team don't have to win (Last strike before winning) , but it crashes 
like described.
```

-----
Original comment by `rumange...@gmail.com` on 2 Jan 2011 at 11:56

**Comment by GoogleCodeExporter** [comment]
```
Thank you for reporting.
I'm not sure if the crash in Comment #1 is actually related to this bug.


I can confirm this bug as still existent in current tip ( rb3e23cb69ab4 )

It happens sometimes when AI is thinking while (last?) enemy hog dies.


Backtrace of the crash:

Program received signal SIGFPE, Arithmetic exception.
#0  0x00000000004dfd6d in FILLTARGETS () at uAIMisc.pas:115
#1  0x00000000004df8f5 in STARTTHINK (ME=0x7fffea3c0180) at uAI.pas:308
#2  0x00000000004dfaee in PROCESSBOT () at uAI.pas:340
#3  0x0000000000469863 in DOGAMETICK (LAG=17) at uGame.pas:54
#4  0x0000000000444304 in DOTIMER (LAG=17) at hwengine.pas:88
#5  0x00000000004449ad in MAINLOOP () at hwengine.pas:216
#6  0x0000000000445400 in GAME () at hwengine.pas:339
#7  0x0000000000447a02 in main () at hwengine.pas:514
```

-----
Original comment by `sheepyluva` on 12 Sep 2011 at 6:02
* Changed state: **Accepted**
* Added labels: **Component-Logic**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in revision 099464aab4b6
```

-----
Original comment by `unC0Rr` on 13 Sep 2011 at 4:11
* Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Verified, game doesn't crash
```

-----
Original comment by `pilarski...@gmail.com` on 29 Nov 2011 at 7:28
* Changed state: **Verified**
